client/ayon_deadline/addon.py

```diff
@@ -67,26 +72,146 @@ def get_deadline_pools(webservice, auth=None, log=None):
+                password
+            log (Optional[Logger]): Logger to log errors to, if provided.
+        Returns:
+            List[str]: Limit Groups.
+        Throws:
+            RuntimeError: If deadline webservice is unreachable.
+
+        """
+        endpoint = "{}/api/groups".format(webservice)
+        return DeadlineAddon._get_deadline_info(
+            endpoint, auth, log, item_type="groups")
+
+    @staticmethod
+    def get_deadline_limit_groups(webservice, auth=None, log=None):
+        """Get Limit Groups from Deadline.
+        Args:
+            webservice (str): Server url.
+             auth (Optional[Tuple[str, str]]): Tuple containing username,
+                password
+            log (Optional[Logger]): Logger to log errors to, if provided.
+        Returns:
+            List[str]: Limit Groups.
+        Throws:
+            RuntimeError: If deadline webservice is unreachable.
+
+        """
+        endpoint = "{}/api/limitgroups?NamesOnly=true".format(webservice)
+        return DeadlineAddon._get_deadline_info(
+            endpoint, auth, log, item_type="limitgroups")
+
+    @staticmethod
+    def get_deadline_workers(webservice, auth=None, log=None):
+        """Get Groups from Deadline.
+        Args:
+            webservice (str): Server url.
+             auth (Optional[Tuple[str, str]]): Tuple containing username,
+                password
+            log (Optional[Logger]): Logger to log errors to, if provided.
+        Returns:
+            List[str]: Limit Groups.
+        Throws:
+            RuntimeError: If deadline webservice is unreachable.
+
+        """
+        endpoint = "{}/api/slaves?NamesOnly=true".format(webservice)
+        return DeadlineAddon._get_deadline_info(
+            endpoint, auth, log, item_type="workers")
+
+    @staticmethod
+    def _get_deadline_info(endpoint, auth=None, log=None, item_type=None):
         from .abstract_submit_deadline import requests_get
 
         if not log:
             log = Logger.get_logger(__name__)
 
-        argument = "{}/api/pools?NamesOnly=true".format(webservice)
         try:
             kwargs = {}
             if auth:
                 kwargs["auth"] = auth
-            response = requests_get(argument, **kwargs)
+            response = requests_get(endpoint, **kwargs)
         except requests.exceptions.ConnectionError as exc:
-            msg = 'Cannot connect to DL web service {}'.format(webservice)
+            msg = 'Cannot connect to DL web service {}'.format(endpoint)
             log.error(msg)
             six.reraise(
                 DeadlineWebserviceError,
                 DeadlineWebserviceError('{} - {}'.format(msg, exc)),
                 sys.exc_info()[2])
         if not response.ok:
-            log.warning("No pools retrieved")
+            log.warning(f"No {item_type} retrieved")
             return []
 
         return response.json()
+
+    def pools_per_server(self, server_name):
```

Comment:
Should we refactor these method names to `pools_by_server`. So _per_ to _by_, that way it matches more with `ayon_api` functions like `get_folder_by_path`.

I wonder even if we should make this `get_pools_by_server_name` to be fully explicit.
Can we also get return type hints for each of these? I suppose each is `List[str]`?

Reply:
Done